server/src/main/java/com/datastrato/gravitino/server/web/rest/CatalogOperations.java

```diff
@@ -145,7 +158,9 @@ public Response dropCatalog(
           httpRequest,
           () -> {
             NameIdentifier ident = NameIdentifier.ofCatalog(metalakeName, catalogName);
-            boolean dropped = manager.dropCatalog(ident);
+            boolean dropped =
+                TreeLockUtils.doWithTreeLock(
+                    ident, LockType.WRITE, () -> manager.dropCatalog(ident));
```

Comment:
Here, I think it should also lock the metalake with write lock, otherwise, the list operation will not be consistent, am I right?

Reply:
`List` catalogs under a metalake need the write lock of the metalake, at the same time, we can't get the write lock of the catalogs to be deleted and vice verse.